Sources/MapboxNavigation/NavigationViewController.swift

```diff
@@ -1103,6 +1103,7 @@ extension NavigationViewController: TopBannerViewControllerDelegate {
     
     public func topBanner(_ banner: TopBannerViewController, didSelect legIndex: Int, stepIndex: Int, cell: StepTableViewCell) {
         let progress = navigationService.routeProgress
+        guard stepIndex < progress.route.legs[legIndex].steps.count else { return }
```

Comment:
Should we also check `legIndex` here? Or to use `safe:` subscript?

Reply:
Thanks for the suggestion, I'll add check to this part as well.